### PR TITLE
Add some null checks to mark functions.

### PR DESCRIPTION
**Pull request by aardvark179**
Added some extra null checks for issues found when running very mark functions very aggressively while running OJ tests.

### CODE REVIEW

File: ext/oj/fast.c [hunk 1]
```diff
@@ -677,21 +677,23 @@ static void free_doc_cb(void *x) {
 }
 
 static void mark_leaf(Leaf leaf) {
-    switch (leaf->value_type) {
-    case COL_VAL:
-        if (NULL != leaf->elements) {
-            Leaf first = leaf->elements->next;
-            Leaf e     = first;
+    if (NULL != leaf) {
```
Comment (ohler55) [inline ext/oj/fast.c:680]: Was there a code change here or just reformatting? Please use the clang formatter in any case.

Comment (aardvark179) [reply ext/oj/fast.c:680]: There is a code change, that initial check that `leaf` isn't null. Clang format leaves my change unaffected, but does touch quite a lot of other lines in that file.

**Comment by ohler55** [comment]
How does the use of DATA_PTR impact performance? Any change?

The NULL check parts are good though.

File: ext/oj/fast.c (hunk 1 above)
Comment (ohler55) [reply ext/oj/fast.c:680]: Odd, the .clang-format file indicates to never use tabs yet I think the diff shows the introduction of tabs unless I am mistaken.

Comment (aardvark179) [reply ext/oj/fast.c:680]: I think it's just showing the different in white space (adding surrounding `if (NULL != leaf` increases the indentation level).

**Comment by aardvark179** [comment]
The added `DATA_PTR` should have a minimal impact on performance. On MRI it's one indirection, and on TruffleRuby it's the only clue we have that you might have accessed the wrapped struct, and that it might need to have its mark function run again.

Ultimately I'd like to submit a patch that avoids using mark functions entirely (because they are relatively expensive to emulate on TruffleRuby), but that still needs a little work.

**Comment by ohler55** [comment]
Interesting to learn that Truffle Ruby tracks by use and not by marking.

I see what you mean about the indentation. I suspect you are correct. I'll approve once the tests complete.

**Review by ohler55 — APPROVED** [review]

**Pull request merged**